packages/grid/x-grid/src/XGrid.test.tsx

```diff
@@ -264,4 +265,59 @@ describe('<XGrid />', () => {
       expect(getColumnValues()).to.deep.equal(['Puma', 'Nike', 'Adidas']);
     });
   });
+  describe('state', () => {
+    it('should allow to control the state using useState', () => {
+      function GridStateTest() {
+        const [gridState] = useState<Partial<GridState>>({
+          sorting: { sortModel: [{ field: 'brand', sort: 'desc' }], sortedRows: [2, 0, 1] },
+        });
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid {...defaultProps} state={gridState} />
+          </div>
+        );
+      }
+
+      render(<GridStateTest />);
+      expect(getColumnValues()).to.deep.equal(['Puma', 'Nike', 'Adidas']);
+    });
+
+    it('should allow to control the state using apiRef', () => {
+      function GridStateTest() {
+        const apiRef = useApiRef();
+        React.useEffect(() => {
+          apiRef.current.setState((prev) => ({
+            ...prev,
+            sorting: { ...prev.sorting, sortModel: [{ field: 'brand', sort: 'asc' }] },
+          }));
+        }, [apiRef]);
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid {...defaultProps} apiRef={apiRef} />
+          </div>
+        );
+      }
+
+      render(<GridStateTest />);
+      expect(getColumnValues()).to.deep.equal(['Adidas', 'Nike', 'Puma']);
+    });
+
+    it('should trigger on state change', () => {
+      let hasStateChanged = false;
+      const onStateChange = () => {
+        hasStateChanged = true;
+      };
+
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <XGrid {...defaultProps} onStateChange={onStateChange} />
```

Comment:
What about we use a spy and we assert that it's called with the right arguments?